app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -33,6 +33,12 @@ cat << EOF > $SSL_CA_CERT
 <%= foreman_server_ca_cert %>
 EOF
 
+if [ -f /usr/bin/hostname ]; then
```

Comment:
I would check that it's executable.
```suggestion
if [ -x /usr/bin/hostname ]; then
```

Reply:
Oh, and on some systems it may not be in `/usr/bin`. I would consider this:
```suggestion
if [ -x $(command -v hostname) ]; then
```

(or at least some solution that looks in `$PATH`.